helios-client/src/main/java/com/spotify/helios/common/descriptors/RollingOperation.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.common.descriptors;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+/**
+ * Represents a Helios deployment group.
+ *
+ * An sample expression of it in JSON might be:
+ * <pre>
+ * {
+ *   "id":"foo-group",
+ *   "job":"foo:0.1.0",
+ *   ],
+ *   "rolloutOptions":{
+ *     "migrate":false,
+ *     "parallelism":2,
+ *     "timeout":1000,
+ *     "overlap":true,
+ *     "token": "insecure-access-token"
+ *   }
+ * }
+ * </pre>
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class RollingOperation extends Descriptor {
+
+  public enum Reason {
+    MANUAL,
+    HOSTS_CHANGED,
+  }
+
+  public static final String EMPTY_ID = "";
+  public static final String EMPTY_DEPLOYMENT_GROUP_NAME = "";
+  public static final JobId EMPTY_JOB_ID = null;
+
+  private final String id;
+  private final String deploymentGroupName;
+  private final JobId jobId;
+  private final Reason reason;
+  private final RolloutOptions rolloutOptions;
+
+  /**
+   * Create a rolling operation.
+   *
+   * @param id The docker id to use.
+   * @param jobId The job ID for the deployment group.
+   */
+  public RollingOperation(
+      @JsonProperty("id") final String id,
+      @JsonProperty("deploymentGroupName") @Nullable final String deploymentGroupName,
```

Comment:
Do we need this to be annotated with `@Nullable`? Isn't this required? Same for job and reason below.